File: DesignPatterns/Strategy/MallardDuck.cpp
#include "MallardDuck.h"



MallardDuck::MallardDuck()
{
	quackBehavior = new Quack();
	flyBehavior = new FlyWithWings();
}


MallardDuck::~MallardDuck()
{
}

void MallardDuck::display()
{
	std::cout << "Mallard Duck!" << std::endl;
}

File: my-adapter/src/main/java/com/designpattern/adapter/MallardDuck.java
package com.designpattern.adapter;

public class MallardDuck implements Duck{
    public MallardDuck(){
    }
    @Override
    public void quack() {
        System.out.println("Quack");
    }
    @Override
    public void fly() {
        System.out.println("Fly");
        
    }
}